### PR TITLE
[Feature](TrinoConnector-Catalog) Fe supports `trino-connector` catalog

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number:  #31719

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -2516,6 +2516,10 @@ public class Config extends ConfigBase {
             options = {"default", "ranger-doris"})
     public static String access_controller_type = "default";
 
+    @ConfField(mutable = true, masterOnly = false, description = {"指定 trino-connector catalog 的插件默认加载路径",
+            "Specify the default plugins loading path for the trino-connector catalog"})
+    public static String trino_connector_plugin_dir = EnvUtils.getDorisHome() + "/lib/connectors";
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2521]:
```suggestion
    public static String trino_connector_plugin_dir = EnvUtils.getDorisHome() + "/connectors";
```

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -780,6 +795,31 @@ public static Env getCurrentEnv() {
         }
     }
 
+    private void initSpiEnvironment() {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:798]: Can it be lazy init?

File: gensrc/thrift/Descriptors.thrift [hunk 3]
```diff
@@ -343,6 +343,12 @@ struct TMCTable {
   8: optional string tunnel_url
 }
 
+struct TTrinoConnectorTable {
+  1: required string db_name
```
Comment (morningman) [inline gensrc/thrift/Descriptors.thrift:347]: use optional

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java (hunk 2 above)
Comment (BePPPower) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:798]: It will be modified in the next pr.

**Review by zddr — APPROVED** [review]

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**